src/msw/printwin.cpp

```diff
@@ -207,9 +232,25 @@ bool wxWindowsPrinter::Print(wxWindow *parent, wxPrintout *printout, bool prompt
             if ( m_printDialogData.GetSelection() && !printout->IsPageSelected(pn) )
                 continue;
 
-            win->SetProgress(pn - minPageNum + 1,
-                             maxPageNum - minPageNum + 1,
-                             copyCount, maxCopyCount);
+            // check ranges, skip pages not in range
+            if (!m_printDialogData.GetAllPages() && !m_printDialogData.GetPageRanges().empty())
+            {
+                bool inRanges = false;
+                for (const wxPrintPageRange& range : m_printDialogData.GetPageRanges())
+                {
+                    if (range.fromPage <= pn && range.toPage >= pn)
+                    {
+                        inRanges = true;
+                        break;
+                    }
+                }
+
+                if (!inRanges)
+                    continue;
+            }
+
+            printCnt += 1;
```

Comment:
A comment explaining why do we do this would be helpful as it's not immediately obvious.